spec/lib/request-options.spec.js

```diff
@@ -0,0 +1,34 @@
+"use strict";
+
+describe("lib/requestOptions", () => {
+    var authToken, libOptions, logger, options, requestOptionsFactory;
+
+    authToken = "abcd1234";
+    logger = require("../../lib/logger")();
+    requestOptionsFactory = () => {
+        return require("../../lib/request-options")(libOptions, logger);
+    };
+
+    describe(".createOptions()", () => {
```

Comment:
I like this test as well. First because it doesn't mock anything but second because of the way the assertions are done. Tests everything without making it hard to maintain.